### ISSUE TITLE
smoosher isn't picking up changes to CSS files when run within a watch task

### ISSUE THREAD

**Issue by Ghodmode** [gabrielflorit/gulp-smoosher#19]
I have a css task that runs some postCSS processors on my CSS file before writing the changes. Then I have an HTML task that runs smoosher followed by [htmlmin](https://www.npmjs.com/package/gulp-htmlmin/).

When I run the tasks separately, it works fine, but if I pass the tasks as an option to the `gulp.watch()` function, the changes in the CSS don't appear in the final file even though they do appear in the file created by the CSS task.

Is smoosher somehow getting the CSS file before the css task finishes writing it?

gulpfile.js:

```js
var gulp = require('gulp');
var postcss = require('gulp-postcss');
var autoprefixer = require('autoprefixer');
var cssnext = require('cssnext');
var precss = require('precss');
var nesting = require('postcss-nesting');
var cssnano = require('cssnano');
var htmlmin = require('gulp-htmlmin');
var smoosher = require('gulp-smoosher');

var paths = {
    css: 'src/*.css',
    html: 'src/*.html'
};

gulp.task('css', function() {
    var processors = [
        nesting,
        autoprefixer,
        cssnext,
        precss,
        cssnano
    ];
    return gulp.src(paths.css)
        .pipe(postcss(processors))
        .pipe(gulp.dest('css'));
});

gulp.task('html', function() {
  return gulp.src('src/*.html')
    .pipe(smoosher({ base: '.' }))
    .pipe(htmlmin({
        collapseWhitespace: true,
        conservativeCollapse: true,
        removeComments: true,
        collapseInlineTagWhitespace: true,
        collapseBooleanAttributes: true,
        removeAttributeQuotes: true,
        removeRedundantAttributes: true,
        removeEmptyAttributes: true,
        removeScriptTypeAttributes: true,
        removeStyleLinkTypeAttributes: true,
        removeOptionalTags: true,
        minifyCSS: true
    }))
    .pipe(gulp.dest('.'))
});

gulp.task('watch', function(){
    gulp.watch(paths.css, ['css', 'html']);
    gulp.watch(paths.html, ['html']);
});
```

**Comment by gabrielflorit** [comment]
@Ghodmode If I remember correctly, gulp tasks don't run in series, they run parallel. Try this instead:

```js
var sequence = require('run-sequence');

gulp.task('css-then-html', function(done) {
  sequence('css', 'html', done);
});

gulp.task('watch', function(){
    gulp.watch(paths.css, ['css-then-html']);
    \\ etc
});

**Comment by Ghodmode** [comment]
Thanks @gabrielflorit That works now. I would swear that I tried that already before I posted this issue, but I must have done something wrong.

Aside from naming the variable `runSequence` (copied and pasted from the [run-sequence package page](https://www.npmjs.com/package/run-sequence)) and not using a callback, I thought my version was the same as your example, but my memory must be faulty.

I wish I could go back and see what I did wrong before but I didn't commit my code using the run-sequence plugin because it wasn't working.

I also didn't realize that there wouldn't be an error if I used a callback argument with the task's function, but didn't pass it a callback.

Thank you very much for your help.
